routers/repo/view.go

```diff
@@ -54,18 +54,38 @@ func renderDirectory(ctx *context.Context, treeLink string) {
 		return
 	}
 
-	var readmeFile *git.Blob
+	var readmes []struct {
+		*git.Blob
+		tp string
+	}
 	for _, entry := range entries {
-		if entry.IsDir() || !markup.IsReadmeFile(entry.Name()) {
+		if entry.IsDir() {
+			continue
+		}
+
+		tp, ok := markup.ReadmeFileType(entry.Name())
+		if !ok {
 			continue
 		}
 
-		// TODO: collect all possible README files and show with priority.
-		readmeFile = entry.Blob()
-		break
+		readmes = append(readmes, struct {
+			*git.Blob
+			tp string
+		}{
+			entry.Blob(),
+			tp,
+		})
 	}
 
-	if readmeFile != nil {
+	if len(readmes) > 0 {
+		var readmeFile *git.Blob
+		for _, readme := range readmes {
```

Comment:
I found that this add another point that it default to even "bad filetype" if not one good is found.
But this could be implemented directly in the first loop.